R/load_forecasts_repo.R

```diff
@@ -0,0 +1,118 @@
+#' Load forecasts from reichlab/covid19-forecast-hub repo.
+#' 
+#' @param file_path path to local clone of the reichlab/covid19-forecast-hub
+#' repository
+#' @param models character vector of model abbreviations.
+#' If missing, forecasts for all models that submitted forecasts 
+#' meeting the other criteria are returned.
+#' @param forecast_dates date vector to load the most recent forecast from
+#' @param locations list of valid fips code. Defaults to all locations with 
+#' available forecasts.
+#' @param types character vector specifying type of forecasts to load: “quantile” 
+#' or “point”. Defaults to c(“quantile”, “point”)
+#' @param targets character vector of targets to retrieve, for example
+#' c('1 wk ahead cum death', '2 wk ahead cum death'). Defaults to all targets.
+#' 
+#' @return data frame with columns model, forecast_date,location, target, 
+#' type, quantile, value, horizon and target_end_date.
+#'
+#' @export
+load_forecasts_repo <- function(file_path, models, forecast_dates, locations, types, targets){
+  
+  # validate models
+  all_valid_models <- covidHubUtils:::get_all_model_abbr()
+  
+  if (!missing(models)){
+    models <- match.arg(models, choices = all_valid_models, several.ok = TRUE)
+  } else {
+    models <- all_valid_models
+  }
+  
+  
+  # validate locations
+  all_valid_fips <- covidHubUtils::hub_locations %>%
+    pull(fips)
+  
+  if (!missing(locations)){
+    locations <- match.arg(locations, choices = all_valid_fips, several.ok = TRUE)
+  } else{
+    locations <- all_valid_fips
+  }
+  
+  # validate types
+  if (!missing(types)){
+    types <- match.arg(types, choices = c("point", "quantile"), several.ok = TRUE)
+  } else {
+    types = c("point", "quantile")
+  }
+  
+  # validate targets 
+  all_valid_targets <- c(
```

Comment:
if these targets ever change, this hard-coded stuff might throw an error. could we retrieve these valid targets straight from a call to zoltar?